### PR TITLE
SAK-47348: Assignment: Webservice for consulting time in worklog to a task

### PR DESCRIPTION
**Pull request by victorGomollon**
[Assignment: Webservice for consulting time in worklog to a task](https://sakaiproject.atlassian.net/browse/SAK-47348)

### CODE REVIEW

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 1]
```diff
@@ -624,6 +624,60 @@ public int removeTimeSheet(Map<String, Object> params) {
...
+ 		if(s.isPresent()) {
+ 			AssignmentSubmissionSubmitter submitter =  s.get();
+ 			if(submitter.getSubmittee()) {
+ 				timeSpent = submitter.getTimeSpent();
+ 			}else {
+ 				timeSpent = assignmentService.getTotalTimeSheet(submitter);
+ 			}
+ 			assignData.put("timeSpent", timeSpent);
+ 		}else {
+ 			throw new EntityException("No existen registros en bitacora para la tarea ", assignmentId, HttpServletResponse.SC_BAD_REQUEST);
```
Comment (juanjmerono) [inline assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:675]: Change to english :)

Comment (victorGomollon) [reply assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java]: Done

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 2]
```diff
@@ -624,6 +624,60 @@ public int removeTimeSheet(Map<String, Object> params) {
...
+ 		Optional<AssignmentSubmissionSubmitter> s = as.getSubmitters().stream().findAny();
+ 		String timeSpent = "";
+ 		if(s.isPresent()) {
+ 			AssignmentSubmissionSubmitter submitter =  s.get();
+ 			if(submitter.getSubmittee()) {
+ 				timeSpent = submitter.getTimeSpent();
+ 			}else {
+ 				timeSpent = assignmentService.getTotalTimeSheet(submitter);
+ 			}
+ 			assignData.put("timeSpent", timeSpent);
```
Comment (mpellicer) [inline assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:673]: submitter.getSubmittee() ?  submitter.getTimeSpent() : assignmentService.getTotalTimeSheet(submitter);

**Review by mpellicer — CHANGES_REQUESTED** [review]
Minor changes, look good so far, well done @victorGomollon

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 3]
```diff
@@ -624,6 +624,60 @@ public int removeTimeSheet(Map<String, Object> params) {
         return HttpServletResponse.SC_OK;
     }
```
[the remark this answers is not in the document]
Comment (victorGomollon) [reply assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:626]: Done

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 4]
```diff
@@ -624,6 +624,60 @@ public int removeTimeSheet(Map<String, Object> params) {
...
+ 		try {
+ 			as = assignmentService.getSubmission(assignmentId, u);
+ 		} catch (PermissionException e1) {
+ 			log.warn("You can't modify this sumbitter");
+ 			throw new EntityException("You don't have permissions read submission " + assignmentId, "", HttpServletResponse.SC_FORBIDDEN);
+ 		}
+ 		if(as == null) {
+ 			throw new EntityException("There are not worklog records for this assignment ", assignmentId, HttpServletResponse.SC_BAD_REQUEST); 
+ 		}
+ 		Optional<AssignmentSubmissionSubmitter> s = as.getSubmitters().stream().findAny();
```
[the remark this answers is not in the document]
Comment (victorGomollon) [reply assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java]: Done

**Review by mpellicer — APPROVED** [review]

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java (hunk 2 above)
Comment (victorGomollon) [reply assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java]: Done

**Pull request merged**